lib/Doctrine/ORM/Query/AST/Functions/IdentityFunction.php

```diff
@@ -61,16 +61,19 @@ public function getSql(SqlWalker $sqlWalker)
         $joinColumn     = reset($assoc['joinColumns']);
 
         if ($this->fieldMapping !== null) {
-            if ( ! isset($targetEntity->fieldMappings[$this->fieldMapping])) {
-                throw new QueryException(sprintf('Undefined reference field mapping "%s"', $this->fieldMapping));
-            }
-
-            $field      = $targetEntity->fieldMappings[$this->fieldMapping];
             $joinColumn = null;
 
+            if (isset($targetEntity->fieldMappings[$this->fieldMapping])) {
+                $columnName = $targetEntity->fieldMappings[$this->fieldMapping]['columnName'];
+            } elseif (isset($targetEntity->associationMappings[$this->fieldMapping])) {
+                $columnName = reset($targetEntity->associationMappings[$this->fieldMapping]['joinColumns'])['name'];
```

Comment:
I think `reset()` is here because we don't support `IDENTITY` with composite keys?

Reply:
I guess this is the reason.

Reply:
In that case could we throw an exception if there are more than one key?

Reply:
Maybe, not sure why it wasn't done like that in the first place. Right now, it will only return the value of the first part of the composite key, right?

Reply:
Hm from https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/dql-doctrine-query-language.html#:~:text=the%20IDENTITY()%20DQL I read

> the IDENTITY() DQL function also works for composite primary keys

So there must be another reason :thinking: